src/main/java/org/georchestra/gateway/filter/headers/providers/UserSecurityHeadersProvider.java

```diff
@@ -0,0 +1,80 @@
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import org.georchestra.gateway.filter.headers.HeaderProvider;
+import org.georchestra.gateway.model.GeorchestraTargetConfig;
+import org.georchestra.gateway.model.GeorchestraUser;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.web.server.ServerWebExchange;
+
+public class UserSecurityHeadersProvider extends HeaderProvider {
+
+    public @Override Consumer<HttpHeaders> prepare(ServerWebExchange exchange) {
+        return headers -> {
+            GeorchestraTargetConfig.getTarget(exchange)//
+                    .map(GeorchestraTargetConfig::headers)//
+                    .ifPresent(mappings -> {
+                        Optional<GeorchestraUser> user = GeorchestraUser.resolve(exchange);
+                        add(headers, "sec-proxy", mappings.getProxy(), "true");
+                        add(headers, "sec-username", mappings.getUsername(), user.map(GeorchestraUser::getUsername));
+                        add(headers, "sec-org", mappings.getOrg(), user.map(GeorchestraUser::getOrganization));
+                        add(headers, "sec-email", mappings.getEmail(), user.map(GeorchestraUser::getEmail));
+                        add(headers, "sec-firstname", mappings.getFirstname(), user.map(GeorchestraUser::getFirstName));
+                        add(headers, "sec-lastname", mappings.getLastname(), user.map(GeorchestraUser::getLastName));
+                        add(headers, "sec-tel", mappings.getTel(), user.map(GeorchestraUser::getTelephoneNumber));
+
+                        List<String> roles = user.map(GeorchestraUser::getRoles).orElse(List.of());
+                        add(headers, "sec-roles", mappings.getRoles(), roles);
+                    });
+        };
+    }
+
+    String resolveName(Principal p) {
+        System.err.println("Resolving " + p.getClass().getName());
+        String name;
+        if (p instanceof OAuth2AuthenticationToken) {
+            OAuth2AuthenticationToken u = ((OAuth2AuthenticationToken) p);
+            Collection<GrantedAuthority> authorities = u.getAuthorities();
+            String authorizedClientRegistrationId = u.getAuthorizedClientRegistrationId();
+            Object credentials = u.getCredentials();
+            Object details = u.getDetails();
+            String n = u.getName();
+            OAuth2User principal = u.getPrincipal();
+            Map<String, Object> attributes = principal.getAttributes();
+            String name2 = principal.getName();
+            Map<String, Object> attributes2 = principal.getAttributes();
+            name = (String) attributes2.get("email");
+        } else {
```

Comment:
I guess the different auth mechanisms (LDAP, AD, ...) will have their own code here